File: Temperatures.java
import java.util.*;
import java.io.*;
import java.math.*;
class Solution {public static void main(String args[]){Scanner in=new Scanner(System.in);int n = in.nextInt();int diff=5527;if(n==0){diff=0;}for(int i=0;i<n;i++){int t=in.nextInt();if(Math.abs(diff)>Math.abs(t)){diff=t;}else if(Math.abs(t)==Math.abs(diff)){if(t>diff)diff=t;}}System.out.println(diff);}}

File: Temperatures.cpp
#include <iostream>
int main(){int n,t,x=5529;std::cin>>n;for(int i=0;i<n;i++){std::cin>>t;(abs(t)<abs(x))?(x=t):(abs(t)==abs(x)&&(t>x))?(x=t):t;}if(!n)x=0;std::cout<<x;}
